R/server.R

```diff
@@ -459,6 +459,47 @@ startApp <- function(appObj, port, host, quiet) {
     ),
     .globals$resourcePaths
   )
+
+  # throw an informative warning if a subdirectory of the
+  # app's www dir conflicts with another resource prefix
+  wwwDir <- httpuvApp$staticPaths[["/"]]$path
+  if (length(wwwDir)) {
+    wwwSubDirs <- list.dirs(wwwDir, recursive = FALSE, full.names = FALSE)
+    resourceConflicts <- intersect(wwwSubDirs, names(httpuvApp$staticPaths))
+    if (length(resourceConflicts)) {
+      warning(
+        "Found subdirectories of your app's www/ directory that ",
+        "conflict with other resource URL prefixes.",
+        "Consider renaming these directories: '",
+        paste("www", resourceConflicts, collapse = "', '"), "'",
+        call. = FALSE
+      )
+    }
+  }
+
+  # check for conflicts in each pairwise combinations of resource mappings
+  checkResourceConflict <- function(paths) {
+    if (length(paths) < 2) return(NULL)
+    # ensure paths is a named character vector: c(resource_path = local_path)
+    local_paths <- vapply(paths, function(x) if (inherits(x, "staticPath")) x$path else x, character(1))
+    paths <- setNames(local_paths, names(paths))
+    # get all possible pairwise combinations of paths
+    pair_indicies <- utils::combn(length(paths), 2, simplify = FALSE)
```

Comment:
Super nitpick: should be spelled "indices". You can ignore this comment if you want though. :)